x/leverage/types/keys.go

```diff
@@ -46,3 +52,47 @@ func CreateRegisteredTokenKey(baseTokenDenom string) []byte {
 	key = append(key, []byte(baseTokenDenom)...)
 	return append(key, 0) // append 0 for null-termination
 }
+
+// CreateLoanKey returns a KVStore key for getting and setting a Loan in a single denom and borrower address
+func CreateLoanKey(borrowerAddr sdk.AccAddress, tokenDenom string) []byte {
+	// loanprefix | lengthprefixed(borrowerAddr) | denom
+	var key []byte
+	key = append(key, KeyPrefixLoanToken...)
+	addr := []byte(borrowerAddr.String())
+	key = append(key, byte(len(addr))) // simple length prefix since len(addr.String()) is always < 255
+	key = append(key, addr...)
+	key = append(key, []byte(tokenDenom)...)
+	return append(key, 0) // append 0 for null-termination
+}
+
+// LoanKeyRange returns start/end keys for creating an iterator over all of an account's open loans.
+func LoanKeyRange(borrowerAddr sdk.AccAddress) ([]byte, []byte) {
+	//	Question: Is this the right way to derive a range for an sdk.Iterator?
+	//
+	//	e.g. if KeyPrefixLoanToken | lengthPrefixed(borrowerAddr.String()) were resolved to
+	//		0x04 | 0x03 0x41 0x42 0x43 (example address string simplified to "ABC")
+	//	then the iterator start/end would be
+	//		0x04 | 0x03 0x41 0x42 0x43 (inclusive start)
+	//		0x04 | 0x03 0x41 0x42 0x44 (exclusive end)
+	//	and keys like the following would fall within the range
+	//		0x04 | 0x03 0x41 0x42 0x43 | ... (any key that has prefix)
+	//
+	//	I couldn't find documentation on this behavior but it seems like
+	//		how it would reasonably work if sdk.Iterators are prefix-friendly.
```

Comment:
```suggestion
	//	Question: Is this the right way to derive a range for an sdk.Iterator?
	//
	//	e.g. if KeyPrefixLoanToken | lengthPrefixed(borrowerAddr.String()) were resolved to
	//		0x04 | 0x03 0x41 0x42 0x43 (example address string simplified to "ABC")
	//	then the iterator start/end would be
	//		0x04 | 0x03 0x41 0x42 0x43 (inclusive start)
	//		0x04 | 0x03 0x41 0x42 0x44 (exclusive end)
	//	and keys like the following would fall within the range
	//		0x04 | 0x03 0x41 0x42 0x43 | ... (any key that has prefix)
	//
	//	I couldn't find documentation on this behavior but it seems like
	//		how it would reasonably work if sdk.Iterators are prefix-friendly.
```
Note: Question